mindsdb/integrations/handlers/neuralforecast_handler/neuralforecast_handler.py

```diff
@@ -0,0 +1,99 @@
+import os
+import string
+import random
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.integrations.handlers.statsforecast_handler.statsforecast_handler import (
+    infer_frequency,
+    transform_to_nixtla_df,
+    get_results_from_nixtla_df,
+)
+from neuralforecast import NeuralForecast
+from neuralforecast.models import NHITS
+from neuralforecast.auto import AutoNHITS
+from ray.tune.search.hyperopt import HyperOptSearch
+
+DEFAULT_FREQUENCY = "D"
+DEFAULT_MODEL_NAME = "NHITS"
+DEFAULT_TRIALS = 20
+MIN_TRIALS_FOR_AUTOTUNE = 3
+
+
+def choose_model(num_trials, horizon, window, exog_vars=[], threshold=MIN_TRIALS_FOR_AUTOTUNE):
+    """Chooses model based on the number of trials allowed by the user.
+
+    A lower args for training time reduces the number of trial models we can
+    search through. Below a certain threshold, we switch to the default
+    NHITS implementation instead of using AutoML to search for the best config.
+    """
+    if num_trials >= threshold:
+        model = AutoNHITS(horizon, gpus=0, num_samples=num_trials, search_alg=HyperOptSearch())
+    else:  # faster implementation without auto parameter tuning
+        model = NHITS(horizon, window, hist_exog_list=exog_vars)
+    return model
+
+
+class NeuralForecastHandler(BaseMLEngine):
+    """Integration with the Nixtla NeuralForecast library for
+    time series forecasting with neural networks.
+    """
+
+    name = "neuralforecast"
+
+    def create(self, target, df, args={}):
+        """Create the NeuralForecast Handler.
+
+        Requires specifying the target column to predict and time series arguments for
+        prediction horizon, time column (order by) and grouping column(s).
+
+        Saves model params to desk, which are called later in the predict() method.
+        """
+        with open("args.txt", "w+") as f:
+            f.write(str(args))
+        time_settings = args["timeseries_settings"]
+        using_args = args["using"]
+        assert time_settings["is_timeseries"], "Specify time series settings in your query"
+        ###### store model args and time series settings in the model folder
+        model_args = {}
+        model_args["target"] = target
+        model_args["horizon"] = time_settings["horizon"]
+        model_args["order_by"] = time_settings["order_by"]
+        model_args["group_by"] = time_settings["group_by"]
+        model_args["frequency"] = (
+            using_args["frequency"] if "frequency" in using_args else infer_frequency(df, time_settings["order_by"])
+        )
+        model_args["model_name"] = DEFAULT_MODEL_NAME
+        num_trials = int(DEFAULT_TRIALS * using_args["train_time"]) if "train_time" in using_args else DEFAULT_TRIALS
+        exog_vars = using_args["exogenous_vars"] if "exogenous_vars" in using_args else []
+
+        # You have to save neuralforecast into a different folder each time.
```

Comment:
As above, opt for `tempfile.mkdtemp` here

Reply:
That's a lot more elegant than my solution - thanks!